astroid/nodes/node_ng.py

```diff
@@ -246,7 +259,7 @@ def frame(self):
         """
         return self.parent.frame()
 
-    def scope(self):
+    def scope(self) -> Optional["LocalsDictNodeNG"]:
```

Comment:
Regarding `LocalsDictNodeNG`: I fear that at some point it will just create another circular dependency. Maybe we should start looking into `Protocol` classes for that 🤔 Haven't really done so myself, thus don't know for certain if it would work.

https://docs.python.org/3/library/typing.html#typing.Protocol

--
Another issue: If at some point some scoped node inherits from another mixin class (not `LocalsDictNodeNG`, this will break.

Reply:
Are you sure? Doesn't using `"Class"` prevent circular dependency?

Reply:
The current code is fine. As you said, using a string as type annotation does prevent the circular dependency.
It's probably just that I'm not really a fan of the class name itself (`LocalsDictNodeNG`). Maybe that should be changed at some point 🤔

Anyway, let's leave it for now.